clifford/numba/_multivector.py

```diff
@@ -55,17 +51,25 @@ def value_type(self):
 def _numba_type_(self):
     layout_type = self.layout._numba_type_
 
-    cache = layout_type._cache
+    relevant_cache = layout_type._c_cache
     dt = self.value.dtype
 
+    if not self.value.flags.c_contiguous:
+        if self.value.flags.f_contiguous:
+            relevant_cache = layout_type._f_cache
+        elif self.value.flags.aligned:
+            relevant_cache = layout_type._a_cache
+        else:
+            raise ValueError("Not sure what to write here")
+
     # now use the dtype to key that cache.
     try:
-        return cache[dt]
+        return relevant_cache[dt]
     except KeyError:
-        # Computing and hashing `dtype_type` is slow, so we do not use it as a
+        # Computing and hashing `value_type` is slow, so we do not use it as a
         # hash key. The raw numpy dtype is much faster to use as a key.
-        dtype_type = _numpy_support.from_dtype(dt)
-        ret = cache[dt] = MultiVectorType(layout_type, dtype_type)
+        value_type = _numpy_support.from_dtype(dt)[:]
```

Comment:
This `[:]` needs to be adjusted to match the order - `[::1]` is C, I forget what `F` is, and `[:]` is any. You could use `numba.types.Array` directly instead I think.

Reply:
https://github.com/numba/numba/blob/3434eac6683a915243d8c4c89d483f9a04fabca9/numba/np/numpy_support.py#L215-L222